core/src/from_meta.rs

```diff
@@ -246,6 +246,53 @@ impl<T: syn::parse::Parse, P: syn::parse::Parse> FromMeta for syn::punctuated::P
     }
 }
 
+/// Parsing support for an array, i.e. "[1 + 2, 2 - 2, 3 * 4]".
+impl FromMeta for syn::ExprArray {
+    fn from_value(value: &Lit) -> Result<Self> {
+        if let Lit::Str(ref ident) = *value {
+            ident
+                .parse::<syn::ExprArray>()
+                .map_err(|_| Error::unknown_lit_str_value(ident))
+        } else {
+            Err(Error::unexpected_lit_type(value))
+        }
+    }
+}
+
+macro_rules! from_number_array {
+    ($ty:ident) => {
+        impl FromMeta for Vec<$ty> {
```

Comment:
I'm a little concerned about introducing `FromMeta` impls for specific types of `Vec`; wouldn't that prevent a blanket impl in the future?

Reply:
Is a blanket impl using something like `std::str::FromStr` (or other std library traits) already blocked due to darling's implementation of `impl FromMeta for Vec<syn::WherePredicate>`?

Reply:
Ah, I think you're right. In that case, I'd like to test how this interacts with `#[darling(multiple)]`, and as long as that works well we should be good.

Can you share a bit more information about the use-case for these?

Reply:
The use case is primarily providing an ergonomic way to provide byte data as an argument to a proc macro. For instance taking slightly modified code from tests/multiple:

`#[hello(ipsum(dolor = 0x12, dolor = 0x34)]
pub struct Baz;
`

It could be rewritten as the following which provides concise code showing the intent of passing an array/vector to the proc macro:
`#[hello(ipsum(dolor = "[0x12, 0x34]")]
pub struct Baz;
`

Whilst it was originally intended for use with a `Vec<u8>`, it was simple to expand via a procedural macro to other unsigned integers so I thought I would. However, mapping the string values from a syn::ExprArray is harder for unsigned integers, as it's returned as an Expr::Unary for negative numbers.

Will you look at interaction with `#[darling(multiple)]` or would you like me to investigate?

Reply:
I ran this test, which worked:

```rust
#[macro_use]
extern crate darling;
#[macro_use]
extern crate syn;

use darling::FromDeriveInput;
use syn::DeriveInput;

#[derive(Debug, FromDeriveInput)]
#[darling(attributes(lorem))]
struct Receiver {
    // #[darling(multiple)]
    value: Vec<u8>,
}

fn main() {
    let di: DeriveInput = parse_quote! {
        #[derive(Receiver)]
        #[lorem(value = "[1, 2]")]
        // #[lorem(value = "1", value = "2")]
        struct Foo;
    };

    dbg!(Receiver::from_derive_input(&di).unwrap());
}
```

Uncommenting the attribute declaration in `Receiver` and swapping which one is used on `Foo` has the desired effects, so I think we can consider this resolved.